OpenRA.Mods.Cnc/Traits/Disguise.cs

```diff
@@ -211,14 +223,58 @@ void HandleDisguise(Player oldEffectiveOwner, bool oldDisguiseSetting)
 
 		void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barrel)
 		{
-			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack))
+			if (info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Attack) || info.RevealDisguiseOn.HasFlag(RevealDisguiseType.Demolish))
```

Comment:
This is wrong.

Demolish triggers the uncloak directly in it's activity and not through an interface,. so i see where this came from. However it doesn't make this any valid - this will render the Demolish value equal to Attack. Either you replace that call in the Demolish activity with an INotifyDemolish interface with this trait also implementing it either you drop this value.